lib/ansible/modules/net_tools/ldap/ldap_attr.py

```diff
@@ -178,40 +149,30 @@
 
 import traceback
 
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.ldap import LdapGeneric, gen_specs, HAS_LDAP
+
 try:
     import ldap
-    import ldap.sasl
-
-    HAS_LDAP = True
```

Comment:
You should keep the `HAS_LDAP = True` here and remove the import for it. The same in all other files.